app/api/speakers.py

```diff
@@ -111,42 +92,68 @@ def before_post(self, args, kwargs, data):
         :return:
         """
         require_relationship(['event', 'user'], data)
-        kwargs['user_id'] = current_identity.id
 
-    def before_create_object(self, data, view_kwargs):
+        if not has_access('is_coorganizer', event_id=data['event']):
+            if has_access('is_user_itself'):
```

Comment:
I'm not sure if `is_user_itself` can be used here, and it checks `kwargs['id'] == user.id`, can you confirm please.